packages/glimmer-runtime/lib/modifier/interfaces.ts

```diff
@@ -4,15 +4,18 @@ import { DynamicScope } from '../environment';
 import { Destroyable } from 'glimmer-util';
 
 export interface ModifierManager<T> {
+  args: EvaluatedArgs;
+  dynamicScope: DynamicScope;
+  create(args: EvaluatedArgs, dynamicScope: DynamicScope): T;
```

Comment:
Actaully, since the element cannot change, we could pass the element to `create` instead, then we don't have pass them to `install` and `update` (consequently, you won't have to buffer the `element` on the env).